crypto/jwx.go

```diff
@@ -338,6 +434,35 @@ func SignatureAlgorithm(key crypto.PublicKey) (jwa.SignatureAlgorithm, error) {
 	}
 }
 
+func EncryptionAlgorithm(key crypto.PublicKey) (jwa.KeyEncryptionAlgorithm, error) {
+	if key == nil {
+		return "", errors.New("no key provided")
+	}
```

Comment:
I'd say move this check to EncryptJWE, make this func private

Reply:
Done